src/ssz/type/container.zig

```diff
@@ -305,6 +320,21 @@ pub fn VariableContainerType(comptime ST: type) type {
             try merkleize(@ptrCast(&chunks), chunk_depth, out);
         }
 
+        /// Creates a new `VariableContainerType` and clones all underlying fields in the container.
+        ///
+        /// Caller owns the memory.
+        pub fn clone(
+            allocator: std.mem.Allocator,
+            value: *const Type,
+        ) !Type {
+            var out: Type = default_value;
+            inline for (fields) |field| {
+                @field(out, field.name) = try field.type.clone(allocator, &@field(value, field.name));
```

Comment:
I prefer using an out parameter which has 2 benefits:
- we don't have to copy value recursively
- consumer can decide either stack allocation if temporary variable, or heap allocation

cc @wemeetagain  I think it's also conform to the current apis we have for example:

```zig
pub fn deserializeFromJson(source: *std.json.Scanner, out: *Type) !void {
```